engine/class_modules/sc_demon_hunter.cpp

```diff
@@ -1040,6 +1043,29 @@ struct soul_fragment_t
       dh->resource_gain( RESOURCE_FURY, dh->spec.blind_faith_fury->effectN( 1 ).resource( RESOURCE_FURY ), dh->gain.blind_faith );
     }
 
+    if ( is_type( soul_fragment::LESSER ) && dh->sets->has_set_bonus( DEMON_HUNTER_VENGEANCE, T28, B4))
```

Comment:
This check is fine (the result will be the same) but as a minor point it may make it easier to search if it was consistent with the other checks in the module which compare against p()->set_bonuses.t28_vengeance_4pc->ok()